app/components/LoginPage.jsx

```diff
@@ -25,9 +25,13 @@ export default class LoginPage extends Component {
 
     onSubmit = () => {
         const { password, username } = this.state;
-        const { location, onLogin } = this.props;
+        const { isSamlEnabled, location, onLogin, samlSsoUrl } = this.props;
         const errors = {};
 
+        if (isSamlEnabled) {
+            window.open(samlSsoUrl, '_self');
```

Comment:
Why not 

```suggestion
            window.location.href = samlSsoUrl;
```

which seems to be the community-preferred option https://stackoverflow.com/questions/4813879/window-open-target-self-v-window-location-href

Reply:
I would do so, that was my original implementation, but had problem with Cypress test and changed it to `window.open`. If you provide me a way to test it with Cypress, I'll change it.

See: https://github.com/cypress-io/cypress/issues/3994
FYI I was testing `experimentalSourceRewriting` without success.

Reply:
I read about the problem for a bit and IMO that's more closely related to https://docs.cypress.io/guides/guides/web-security.html#JavaScript-Redirects rather than some URL rewriting. The problem stems from going cross-origin, which looks not to be supported in Cypress.

To use `window.location.href`, a change is necessary in all tests to make `ssoUrl` be in the same origin as the initial page visited. Thus, for example in the `provides SSO login button when SAML is enabled` test in `login_spec.js`:
1. Change `ssoUrl`, e.g. to `/sso-redirect`
2. Instead of asserting that `window.open` was called, do: `cy.url().should('include', ssoUrl);`

Worked on my end:
![image](https://user-images.githubusercontent.com/889383/104935058-1023e680-59ab-11eb-974b-398f571386f4.png)

Reply:
Good idea. Implemented. 
I used `window.location` instead of `window.location.href` to keep it consistent with other places in the code.

From my perspective in the future we could consider moving to `location.assign(...)` as the location change is more like an action ( => function call) and not just a state change ( => variable assignment).

Reply:
Sure thing :smile: Today I learned! Thanks for sharing about `location.assign`, I did not know such a thing existed. I definitely agree with your reasoning here :slightly_smiling_face: